query/src/sql/exec/util.rs

```diff
@@ -28,3 +34,35 @@ pub fn check_physical(expression: &SExpr) -> bool {
 
     true
 }
+
+/// Format the display name and index of a column into `"{display_name}"_index` format.
+pub fn format_field_name(display_name: &str, index: IndexType) -> String {
+    format!("\"{}\"_{}", display_name, index)
+}
+
+lazy_static! {
+    static ref FIELD_NAME_RE: Regex = Regex::new("\"([^\"]*)\"_([0-9]+)").unwrap();
+}
+
+/// Decode a field name into display name and index
+pub fn decode_field_name(field_name: &str) -> Result<(String, IndexType)> {
+    let result = FIELD_NAME_RE.captures(field_name);
+    match result {
+        Some(res) => {
+            if res.len() != 3 {
+                Err(ErrorCode::LogicalError(format!(
+                    "Invalid field name: {}",
+                    field_name
+                )))
+            } else {
+                let name = res.get(1).unwrap().as_str().to_string();
```

Comment:
replace `unwrap` with `?`

Reply:
https://docs.rs/regex/latest/regex/struct.Captures.html#method.get

`Captures::get` actually returns a `Option<Match>`.

Since the length of `res` has been checked here, it's safe to `unwrap` the `Option`.

Reply:
If the length has been checked, how about use `res[1]` instead?

Reply:
If the length has been checked, how about using `res[1]` instead?

Reply:
If the length has been checked, how about use `res[1]` instead?

Reply:
If the length has been checked, how about using `res[1]` instead?

Reply:
If the length has been checked, how about using `res[1]` instead?

Reply:
If the length has been checked, how about use `res[1]` instead?

Reply:
> If the length has been checked, how about using `res[1]` instead?

Good idea. I haven't notice that `Captures` implemented `Index` trait.